wsd/FileServer.cpp

```diff
@@ -994,6 +994,7 @@ constexpr char BRANDING_UNSUPPORTED[] = "branding-unsupported";
 
 static const std::string ACCESS_TOKEN = "%ACCESS_TOKEN%";
 static const std::string ACCESS_TOKEN_TTL = "%ACCESS_TOKEN_TTL%";
+static const std::string ACCESS_HEADER = "%ACCESS_HEADER%";
```

Comment:
I'm surprised to see a somewhat non-uniform substitution approach; as well as all the complexity in the parser to handle weird cases.

Lets merge this - but I would really prefer to see a uniform parsing of the file for a simple '%%' and then up to the next '%%' is the variable name, and we look that up straight without this extra decoration etc. =)

Is that do-able; it will require tweaking cool.html of course - and anyhow pleased to see Poco for the chop =)

Reply:
Yes. I started with the same sentiments and goals. Unfortunately, the way variables are embedded is different, and I think it's fixable. But I didn't want to change current approach while writing the tokenizer.

I think the issue is that the variables that are embedded in JavaScript need to be in comments, until and unless they are replaced with something valid. But since not all variables are guaranteed to have replacement (since they come from config and elsewhere), their placeholders are inside `<!-- -->` comments.

The solution could've been as simple as always defining variables to a certain safe default value, such that if they aren't defined, we can still replace them blindly and get valid code. That would require moving those JS variables out of comments, though, and I didn't want to make too many changes.

I'm happy to apply the above suggestion and simplify the tokenizer as a follow up.

Reply:
Sounds good - lets clean that up in master - the way we handle defaults is not so pleasant there =)